pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/FunctionsBase.java

```diff
@@ -276,6 +276,26 @@ public FunctionState getFunctionState(final @PathParam("tenant") String tenant,
         return functions.getFunctionState(tenant, namespace, functionName, key, clientAppId(), clientAuthData());
     }
 
+    @POST
+    @ApiOperation(
+            value = "Put the state associated with a Pulsar Function"
+    )
+    @ApiResponses(value = {
+            @ApiResponse(code = 400, message = "Invalid request"),
+            @ApiResponse(code = 403, message = "The requester doesn't have admin permissions"),
+            @ApiResponse(code = 404, message = "The function does not exist"),
+            @ApiResponse(code = 500, message = "Internal server error")
+    })
+    @Path("/{tenant}/{namespace}/{functionName}/state/{key}")
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    public void putFunctionState(final @PathParam("tenant") String tenant,
+                                 final @PathParam("namespace") String namespace,
+                                 final @PathParam("functionName") String functionName,
+                                 final @PathParam("key") String key,
+                                 final @FormDataParam("state") String stateJson) {
```

Comment:
just this as FunctionState, you don't need to get it as a string.  Jersey will automatically deserialize that for you

Reply:
Done